pkg/kubelet/nodelease/controller.go

```diff
@@ -60,27 +60,18 @@ type controller struct {
 }
 
 // NewController constructs and returns a controller
-func NewController(clock clock.Clock, client clientset.Interface, holderIdentity string, leaseDurationSeconds int32, nodeStatusUpdateFrequency time.Duration, onRepeatedHeartbeatFailure func()) Controller {
+func NewController(clock clock.Clock, client clientset.Interface, holderIdentity string, leaseDurationSeconds int32, onRepeatedHeartbeatFailure func()) Controller {
 	var leaseClient coordclientset.LeaseInterface
 	if client != nil {
 		leaseClient = client.CoordinationV1().Leases(corev1.NamespaceNodeLease)
 	}
-	renewInterval := defaultRenewInterval
-	// Users are able to decrease the timeout after which nodes are being
-	// marked as "Ready: Unknown" by NodeLifecycleController to values
-	// smaller than defaultRenewInterval. Until the knob to configure
-	// lease renew interval is exposed to user, we temporarily decrease
-	// renewInterval based on the NodeStatusUpdateFrequency.
-	if renewInterval > nodeStatusUpdateFrequency {
-		renewInterval = nodeStatusUpdateFrequency
-	}
-
+	renewIntervalSeconds := int64(renewIntervalFraction * float64(leaseDurationSeconds))
```

Comment:
It's not what we want - if you set say leaseDurationSecond=3, then this one will be 0 (which definitely isn't what we want).
Should be something like:
```
leaseDuration := time.Duration(leaseDurationSeconds) * time.Second
renewInterval := time.Duration(float64(leaseDuration) * renewIntervalFraction)
``

Reply:
done, thanks